api/src/main/java/org/pmiops/workbench/elasticsearch/ElasticFilters.java

```diff
@@ -157,31 +165,99 @@ private QueryBuilder searchGroupToFilter(SearchGroup sg) {
         }
       }
 
-      // TODO(calbach): Handle Attribute modifiers here, e.g. value_as_number < 10.
-      // TODO(calbach): Handle non-code values, e.g. question answers here.
+      // TODO(freemabd): Handle Blood Pressure and Deceased
       for (SearchParameter param : sgi.getSearchParameters()) {
         String conceptField = "events." + (isStandardConcept(param) ? "concept_id" : "source_concept_id");
-        BoolQueryBuilder b = QueryBuilders.boolQuery()
-            .filter(QueryBuilders.termsQuery(conceptField, toleafConceptIds(ImmutableList.of(param))));
+        if (isNonNestedSchema(param)) {
+          conceptField = nonNestedFields.get(param.getSubtype());
+        }
+        Set<String> leafConceptIds = toleafConceptIds(ImmutableList.of(param));
+        BoolQueryBuilder b = QueryBuilders.boolQuery();
+        if (!leafConceptIds.isEmpty()) {
+          b.filter(QueryBuilders.termsQuery(conceptField, leafConceptIds));
+        }
+        for (Attribute attr : param.getAttributes()) {
+          b.filter(attributeToQuery(attr));
+        }
         for (QueryBuilder f : modFilters) {
           b.filter(f);
         }
-        // "should" gives us "OR" behavior so long as we're in a filter context, which we are. This
-        // translates to N occurrences of criteria 1 OR N occurrences of criteria 2, etc.
-        sgiFilter.should(
+
+        if (isNonNestedSchema(param)) {
+          // setup non nested filter with proper field
+          sgiFilter.should(b);
+        } else {
+          // "should" gives us "OR" behavior so long as we're in a filter context, which we are. This
+          // translates to N occurrences of criteria 1 OR N occurrences of criteria 2, etc.
+          sgiFilter.should(
             QueryBuilders.functionScoreQuery(
-                QueryBuilders.nestedQuery(
-                    // We sum a constant score for each matching document, yielding the total number
-                    // of matching nested documents (events).
-                    "events", QueryBuilders.constantScoreQuery(b), ScoreMode.Total))
-                .setMinScore(occurredAtLeast));
+              QueryBuilders.nestedQuery(
+                // We sum a constant score for each matching document, yielding the total number
+                // of matching nested documents (events).
+                "events", QueryBuilders.constantScoreQuery(b), ScoreMode.Total))
+              .setMinScore(occurredAtLeast));
+        }
       }
-      filter.filter(sgiFilter);
+      //SGI should "OR"
+      filter.should(sgiFilter);
```

Comment:
Oops, thanks

Reply:
I'm thinking with this change that `sgiFilter` is no longer necessary. It's basically:

```
filter = sgiFilter1 || sgiFilter2 || ...
sgiFilter = nestedFilter1 || nestedFilter2 || ... || nonNestedFilter1 || nonNestedFilter2 || ...
```

Reply:
No need to include that in this PR unless you feel inspired. I think the change would just be:
- Remove `sgiFilter`
- Replace existing mentions of `sgiFilter` with `filter`